trees/trees.go

```diff
@@ -109,18 +110,33 @@ func Hash(tree *trillian.Tree) (crypto.Hash, error) {
 	return crypto.SHA256, fmt.Errorf("unexpected hash algorithm: %s", tree.HashAlgorithm)
 }
 
-// Hasher returns a merkle.TreeHasher of the kind configured for the tree.
-func Hasher(tree *trillian.Tree) (merkle.TreeHasher, error) {
+// LogHasher returns a merkle.LogHasher of the kind configured fore the tree.
+func LogHasher(tree *trillian.Tree) (merkle.LogHasher, error) {
 	hash, err := Hash(tree)
 	if err != nil {
 		return nil, err
 	}
 
 	switch tree.HashStrategy {
 	case trillian.HashStrategy_RFC_6962:
-		return rfc6962.TreeHasher{Hash: hash}, nil
+		return rfc6962.New(hash), nil
+	case trillian.HashStrategy_MAP_HASHER:
```

Comment:
This is going to get messy if we have a lot of different algorithms. It would be good if people could use their own without having to modify the core code.